language-support/ts/codegen/tests/ts/daml-ledger-fetch/src/index.ts

```diff
@@ -184,22 +184,29 @@ class Ledger {
   /**
    * Exercise a choice on a contract.
    */
-  async exercise<T, C>(choice: Choice<T, C>, contractId: ContractId<T>, argument: C): Promise<Event<unknown>[]> {
+  async exercise<T, C, R>(choice: Choice<T, C, R>, contractId: ContractId<T>, argument: C): Promise<[R , Event<unknown>[]]> {
     const payload = {
       templateId: choice.template().templateId,
       contractId,
       choice: choice.choiceName,
       argument,
     };
+
     const json = await this.submit('command/exercise', payload);
-    return jtv.Result.withException(jtv.array(decodeEventUnknown()).run(json));
+    console.log ("exercise json response" + JSON.stringify(json));
+    // Decode the server response into a tuple.
+    const responseDecoder: jtv.Decoder<{exerciseResult: R; contracts: Event<unknown>[]}> = jtv.object ({exerciseResult: choice.resultDecoder(), contracts: jtv.array(decodeEventUnknown())});
```

Comment:
This line is too long even for my standards. :)
```suggestion
    const responseDecoder: jtv.Decoder<{exerciseResult: R; contracts: Event<unknown>[]}> = jtv.object({
      exerciseResult: choice.resultDecoder(),
      contracts: jtv.array(decodeEventUnknown()),
    });
```